### ISSUE TITLE
Conflict with Dagger 2.17

### ISSUE THREAD

**Issue by donglua** [airbnb/epoxy#511]
```
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
e: 错误: Epoxy Processor Exception:  Caused by ClassCastException: com.sun.tools.javac.code.Symtab$4 cannot be cast to javax.lang.model.element.TypeElement       Stacktrace:      com.airbnb.epoxy.ControllerModelField.<init>(ControllerModelField.kt:12)       com.airbnb.epoxy.ControllerProcessor.buildFieldInfo(ControllerProcessor.java:223)       com.airbnb.epoxy.ControllerProcessor.addFieldToControllerClass(ControllerProcessor.java:146)       com.airbnb.epoxy.ControllerProcessor.process(ControllerProcessor.java:60)       com.airbnb.epoxy.EpoxyProcessor.processRound(EpoxyProcessor.java:195)       com.airbnb.epoxy.EpoxyProcessor.process(EpoxyProcessor.java:165)
```

Dagger 2.16 works fine.

**Comment by FrantisekGazo** [comment]
why is this closed?

**Comment by donglua** [comment]
https://github.com/google/dagger/issues/1253#issuecomment-413968882

**Comment by elihart** [comment]
Thanks for sharing that link, I'll update Epoxy's version to 1.11.1 as well.
```
Update: you can fix this bug by explicitly depending on JavaPoet 1.11.1 in the annotation processor scope. There was a bug fixed from 1.11.0 but it looks like Dagger needs to be updated
```
